tests/write-json.test.js

```diff
@@ -0,0 +1,91 @@
+const { promises: fs } = require('fs');
+const { convertToJson } = require('../scripts/utils');
+const writeJSON = require("../scripts/utils/write-json");
+const {yamlString, jsonObject} = require("./fixtures/utilsData");
+
+jest.mock('fs', () => ({
+  promises: {
+    readFile: jest.fn(),
+    writeFile: jest.fn(),
+  },
+}));
+
+jest.mock('../scripts/utils', () => ({
+  convertToJson: jest.fn(),
+}));
+
+describe('writeJSON', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should read a file, convert it to JSON, and write the JSON to another file', async () => {
+    const readPath = 'config/testInput.yaml';
+    const writePath = 'config/testOutput.json';
+
+    fs.readFile.mockResolvedValue(yamlString);
+    convertToJson.mockReturnValue(jsonObject);
+
+    await writeJSON(readPath, writePath);
+
+    expect(fs.readFile).toHaveBeenCalledWith(readPath, 'utf-8');
+    expect(convertToJson).toHaveBeenCalledWith(yamlString);
+    expect(fs.writeFile).toHaveBeenCalledWith(writePath, JSON.stringify(jsonObject));
+  });
+
+  test('should log an error and throw if reading the file fails', async () => {
+    const readPath = 'config/testInput.yaml';
+    const writePath = 'config/testOutput.json';
+    const error = new Error('File read error');
+
+    fs.readFile.mockRejectedValue(error);
+    const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    await expect(writeJSON(readPath, writePath)).rejects.toThrow(error);
+
+    expect(consoleErrorSpy).toHaveBeenCalledWith(
+      `Error reading file from path ${readPath}:`,
+      error
+    );
```

Comment:
@anshgoyalevil @sambhavgupta0705 Instead of checking for console messages, can't we return the function from catch block as `return Promise.reject(err);` and then check for the returned value from the function. It is more conventional way to check for errors inside functions.

Reply:
Yeah. I have [suggested](https://github.com/asyncapi/website/pull/3012#issuecomment-2171047555) the use of `throw new Error` for throwing errors and use of try catch block for checking the error thrown. Also, there are many `console.error` statements inside the codebase, which are supposed to be used for debugging purpose, and should not be present in production code.

Reply:
Yeah, @vishvamsinh28  you should refactor the code as well, like along with `console.error` log messages, the function should return the error object from the catch block as well. And you can easily detect those errors inside tests, even if you have manually stubbed the error for a certain function.

Reply:
@akshatnema yea will update it and push changes soon after fixing some errors 👍👍